pkg/virt-operator/resource/apply/rbac.go

```diff
@@ -34,49 +34,50 @@ func (r *Reconciler) createOrUpdateRoleBinding(rb *rbacv1.RoleBinding, imageTag
 	return r.createOrUpdate(rb, imageTag, imageRegistry, id, true)
 }
 
-func (r *Reconciler) createOrUpdate(role runtime.Object,
+func (r *Reconciler) createOrUpdate(required runtime.Object,
 	imageTag, imageRegistry, id string,
 	avoidIfServiceAccount bool) (err error) {
 
-	roleTypeName := role.GetObjectKind().GroupVersionKind().Kind
-	createRole := r.getCreateFunction(role)
-	updateRole := r.getUpdateFunction(role)
+	roleTypeName := required.GetObjectKind().GroupVersionKind().Kind
 
-	cachedRoleInterface, exists, _ := r.getCache(role).Get(role)
-	roleMeta := getMetaObject(role)
-	if avoidIfServiceAccount && !r.stores.ServiceMonitorEnabled && (roleMeta.Name == rbac.MONITOR_SERVICEACCOUNT_NAME) {
+	cachedRoleInterface, exists, _ := r.getCache(required).Get(required)
+	requiredMeta := getMetaObject(required)
+	if avoidIfServiceAccount && !r.stores.ServiceMonitorEnabled && (requiredMeta.Name == rbac.MONITOR_SERVICEACCOUNT_NAME) {
```

Comment:
not sure I understand this variable. `avoidIfServiceAccount` service account is a resource type, but this var is passed in as true for the resource type RoleBinding and Role?

Reply:
I saw that we already began to [discussed this](https://github.com/kubevirt/kubevirt/pull/5224#discussion_r600294351) in my last reconciliation [PR](https://github.com/kubevirt/kubevirt/pull/5224).

The following condition to skip reconciliation of `"kubevirt-monitoring"` resources existed for `Role` and `RoleBinding` before I made any changes to rbac:
```
if !r.stores.ServiceMonitorEnabled && (rb.Name == rbac.MONITOR_SERVICEACCOUNT_NAME) {
	return nil
}
```
Do you think there's a reason to remove this condition?

Anyways, I understand your confusion. I think one plausible solution is to add a private `toSkipReconciliation(obj)` function. There any conditions to skip the reconciliation can be defined. What do you think?

Reply:
I don't think that's needed. I think you can remove this check and instead add it before line  30 && 34. We call this function for each one individually and we do nothing if its true so I'd check before calling the function and remove the func param.